File: corda/java/contracts/src/main/java/com/helloBlock/states/helloBlockState.java
package com.helloBlock.states;

import com.helloBlock.contracts.helloBlockContract;
import com.google.common.collect.ImmutableList;
import net.corda.core.contracts.BelongsToContract;
import net.corda.core.contracts.ContractState;
import net.corda.core.identity.AbstractParty;
import net.corda.core.serialization.ConstructorForDeserialization;
import net.corda.core.serialization.CordaSerializable;
import java.util.List;

@CordaSerializable
@BelongsToContract(helloBlockContract.class)
public class helloBlockState implements ContractState {
    public AbstractParty origin;
    public AbstractParty target;
    public String helloBlock;

    public AbstractParty getOrigin() {
        return origin;
    }

    public AbstractParty getTarget() {
        return target;
    }

    public String getHelloBlock() { return helloBlock; }

    @ConstructorForDeserialization
    public helloBlockState(AbstractParty origin, AbstractParty target, String helloBlock){
        this.origin = origin;
        this.target = target;
        this.helloBlock = helloBlock;
    }

    public helloBlockState(AbstractParty origin, AbstractParty target){
        this(origin,target,"Hello, Block!");
    }

    @Override
    public String toString(){
        return origin.nameOrNull() + "helloBlock";
    }

    @Override
    public List<AbstractParty> getParticipants() {
        return  ImmutableList.of(target);
    }

}

File: corda/kotlin/contracts/src/main/kotlin/com/helloBlock/states/helloBlockState.kt
package com.helloBlock.states

import com.helloBlock.contracts.helloBlockContract
import net.corda.core.contracts.BelongsToContract
import net.corda.core.contracts.ContractState
import net.corda.core.identity.Party

// *********
// * State *
// *********
@BelongsToContract(helloBlockContract::class)
data class helloBlockState(val origin: Party,
                   val target: Party,
                   val helloBlock: String = "Hello, Block!") : ContractState {
    override val participants get() = listOf(target)
}
